pkg/process/keb_processing.go

```diff
@@ -0,0 +1,207 @@
+			p.namedLogger().With(log.KeyResult, log.ValueFail).With(log.KeyError, err.Error()).
+				Error("get runtimes from KEB")
+			time.Sleep(p.KEBClient.Config.PollWaitDuration)
+
+			continue
+		}
+
+		p.namedLogger().Debugf("num of runtimes are: %d", runtimesPage.Count)
+		p.populateCacheAndQueue(runtimesPage)
+		p.namedLogger().Debugf("length of the cache after KEB is done populating: %d", p.Cache.ItemCount())
+		p.namedLogger().Infof("waiting to poll KEB again after %v....", p.KEBClient.Config.PollWaitDuration)
+		recordItemsInCache(float64(p.Cache.ItemCount()))
+		time.Sleep(p.KEBClient.Config.PollWaitDuration)
+	}
+}
+
+// populateCacheAndQueue populates Cache and Queue with new runtimes and deletes the runtimes which should not be tracked.
+func (p *Process) populateCacheAndQueue(runtimes *kebruntime.RuntimesPage) {
+	// clear the gauge to fill it with the new data
+	kebFetchedClusters.Reset()
+
+	validSubAccounts := make(map[string]bool)
+
+	for _, runtime := range runtimes.Data {
+		if runtime.SubAccountID == "" {
+			continue
+		}
+
+		validSubAccounts[runtime.SubAccountID] = true
+		recordObj, isFoundInCache := p.Cache.Get(runtime.SubAccountID)
+
+		// Get provisioning and deprovisioning states if available otherwise return empty string for logging.
+		provisioning := getOrDefault(runtime.Status.Provisioning, "")
+		deprovisioning := getOrDefault(runtime.Status.Deprovisioning, "")
+		p.namedLoggerWithRuntime(runtime).
+			With(log.KeyRuntimeState, runtime.Status.State).
+			With(log.KeyProvisioningStatus, provisioning).
+			With(log.KeyDeprovisioningStatus, deprovisioning).
+			Debug("Runtime state")
+
+		if isRuntimeTrackable(runtime) {
+			newRecord := kmccache.Record{
+				SubAccountID:    runtime.SubAccountID,
+				RuntimeID:       runtime.RuntimeID,
+				InstanceID:      runtime.InstanceID,
+				GlobalAccountID: runtime.GlobalAccountID,
+				ShootName:       runtime.ShootName,
+				ProviderType:    strings.ToLower(runtime.Provider),
+				KubeConfig:      "",
+				Metric:          nil,
+			}
+
+			// record kebFetchedClusters metric for trackable cluster
+			recordKEBFetchedClusters(
+				trackableTrue,
+				runtime.ShootName,
+				runtime.InstanceID,
+				runtime.RuntimeID,
+				runtime.SubAccountID,
+				runtime.GlobalAccountID)
+
+			// Cluster is trackable but does not exist in the cache
+			if !isFoundInCache {
+				err := p.Cache.Add(runtime.SubAccountID, newRecord, cache.NoExpiration)
+				if err != nil {
+					p.namedLoggerWithRecord(&newRecord).With(log.KeyResult, log.ValueFail).With(log.KeyError, err.Error()).Error("Failed to add subAccountID to cache. Skipping queueing it")
+					continue
+				}
+
+				p.Queue.Add(runtime.SubAccountID)
+				p.namedLoggerWithRecord(&newRecord).With(log.KeyResult, log.ValueSuccess).Debug("Queued and added to cache")
+
+				continue
+			}
+
+			// Cluster is trackable and exists in the cache
+			if record, ok := recordObj.(kmccache.Record); ok {
+				if record.ShootName != runtime.ShootName {
```

Comment:
Let's inverse the logic here to reduce nesting
```suggestion
				if record.ShootName == runtime.ShootName {
				  continue
				}
```